wallet/src/repository/wallet/mod.rs

```diff
@@ -50,6 +50,75 @@ impl<T> Wallet<T>
 where
     T: Database,
 {
+    /// Generate transient addresses for synchronization purposes
+    /// This function only creates and inserts addreses
+    pub fn generate_transient_addresses(
+        &self,
+        external_addresses: u16,
+        internal_addresses: u16,
+    ) -> Result<()> {
+        let mut state = self.state.write()?;
+
+        self._generate_transient_addresses(&mut state, external_addresses, internal_addresses)
+    }
+
+    /// Non-locking version of `generate_transient_addresses`
+    pub fn _generate_transient_addresses(
+        &self,
+        state: &mut State,
+        external_addresses: u16,
+        internal_addresses: u16,
+    ) -> Result<()> {
+        // Compute ranges for external and internal addresses
+        let external_range =
+            state.next_external_index..state.next_external_index + u32::from(external_addresses);
+        let internal_range =
+            state.next_internal_index..state.next_internal_index + u32::from(internal_addresses);
+
+        // Generate external addresses
+        for index in external_range {
+            let account = state.account;
+            let keychain = constants::EXTERNAL_KEYCHAIN;
+            let parent_key = &state.keychains[keychain as usize].clone();
+
+            let (address, _) =
+                self.gen_address(None, parent_key, account, keychain, index, false)?;
```

Comment:
This will trigger a deadlock

```suggestion
                self._gen_address(None, parent_key, account, keychain, index, false)?;
```

Reply:
`_gen_address` does not exist :) Maybe in the past

Reply:
Ah sorry, this is not the `gen_address` that uses the state mutex, it's a different function with the same name